src/pymanifold.py

```diff
@@ -125,53 +125,56 @@ def channel(self,
             self.dg.edges[port_from, port_to][key] = attr
         return
 
-    def retrieve(port_in, port_out, attr):
-      if(port_out == "None"):
-        return self.dg.edges[port_in][attr]
+    def retrieve(port_in, attr, port_out=None):
+      if not port_out:
+        if isinstance(port_in, list):
```

Comment:
This should actually be tuple, list would work but when networkx prints an edge it uses a tuple, and it is also what you use in line 135 below.